optimum/onnxruntime/modeling_ort.py

```diff
@@ -0,0 +1,681 @@
+            config=onnx_config,
+            opset=13,
+            output=save_dir.joinpath(ONNX_WEIGHTS_NAME),
+        )
+        kwargs["config"] = model.config.__dict__
+        # 3. load normal model
+        return cls._from_pretrained(save_dir.as_posix(), **kwargs)
+
+
+FEAUTRE_EXTRACTION_SAMPLE = r"""
+    Example of feature extraction:
+
+    ```python
+    >>> from transformers import {processor_class}
+    >>> from optimum.onnxruntime import {model_class}
+    >>> import torch
+
+    >>> tokenizer = {processor_class}.from_pretrained("{checkpoint}")
+    >>> model = {model_class}.from_pretrained("{checkpoint}")
+
+    >>> inputs = tokenizer("My Name is Philipp and i live in Germany.", return_tensors="pt")
+
+    >>> outputs = model(**inputs)
+    >>> logits = outputs.logits
+    >>> list(logits.shape)
+    ```
+
+    Example using `transformers.pipelines`:
+
+    ```python
+    >>> from transformers import {processor_class}, pipeline
+    >>> from optimum.onnxruntime import {model_class}
+
+    >>> tokenizer = {processor_class}.from_pretrained("{checkpoint}")
+    >>> model = {model_class}.from_pretrained("{checkpoint}")
+    >>> onnx_ner = pipeline("feature-extraction", model=model, tokenizer=tokenizer)
+
+    >>> text = "My Name is Philipp and i live in Germany."
+    >>> pred = onnx_ner(text)
+    ```
+"""
+
+
+@add_start_docstrings(
+    """
+    Onnx Model with a MaskedLMOutput for feature-extraction tasks.
+    """,
+    ONNX_MODEL_START_DOCSTRING,
+)
+class OnnxForFeatureExtraction(OnnxModel):
+    """
+    Feature Extraction model for ONNX.
+    """
+
+    # used in from_transformers to export model to onnx
+    pipeline_task = "default"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # create {name:idx} dict for model outputs
+        self.model_outputs = {output_key.name: idx for idx, output_key in enumerate(self.model.get_outputs())}
+
+    @add_start_docstrings_to_model_forward(
+        ONNX_INPUTS_DOCSTRING.format("batch_size, sequence_length")
+        + FEAUTRE_EXTRACTION_SAMPLE.format(
+            processor_class=_TOKENIZER_FOR_DOC,
+            model_class="OnnxForFeatureExtraction",
+            checkpoint="optimum/all-MiniLM-L6-v2",
+        )
+    )
+    def forward(
+        self,
+        input_ids: Optional[torch.Tensor] = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        token_type_ids: Optional[torch.Tensor] = None,
+        **kwargs,
+    ):
+        # converts pytorch inputs into numpy inputs for onnx
+        onnx_inputs = {
+            "input_ids": input_ids.cpu().detach().numpy(),
+            "attention_mask": attention_mask.cpu().detach().numpy(),
+        }
+        if token_type_ids is not None:
+            onnx_inputs["token_type_ids"] = token_type_ids.cpu().detach().numpy()
+        # run inference
+        outputs = self.model.run(None, onnx_inputs)
+        # converts output to namedtuple for pipelines post-processing
```

Comment:
Could we have a `return_dict` parameter? Just a suggestion, I don't know how useful it would be.